### DISCUSSION TITLE
Getting connman to play nicely with k3s

### DISCUSSION THREAD

**Discussion by svector** [k3s-io/k3s#7843] [Ideas]
Hi,
Just came across an issue installing k3s on debian bookworm with LXQT - LXQT recommends connman not network manager, and connman attempts to connect over all the veth* devices created for pods. It creates default routes down each of these interfaces which breaks routing for the host.
You can tell connman to ignore interfaces -see https://wiki.archlinux.org/title/ConnMan#Blacklist_interfaces, so I'm thinking the k3s install could amend connman config to ignore veth* interfaces if it detects connman is installed. Does that seem sensible?
If that makes sense I will see if I can work up a PR for it...

**Comment by brandond** [comment]
> connman attempts to connect over all the veth* devices created for pods

This sounds like it'd be an issue on any host running containers, not just K3s. Does connman not include a sensible default blocklist that ignores common virtual interface patterns?

>the k3s install could amend connman config to ignore veth* interfaces

I don't think it is appropriate for K3s to drop configuration for other packages, so this is not likely something we would do.

**Reply by svector** [reply]
You are correct - it does affect docker & vms too. The debian package doesn't provide an exclusion but there has been a bug raised to correct that  [here](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1023865)
Fair point that k3s shouldn't change config for another package - do you think its worth adding a mention of it to the prerequisites for debian? It took me a couple of days to figure out which component was causing the issue.

**Reply by brandond** [reply]
The fact that this affects so many projects makes me reluctant to document it on our side. It feels like something that should be documented in connman, since it is what ships with unusable defaults.
